linkerd/app/src/lib.rs

```diff
@@ -155,28 +149,24 @@ where
     where
         F: Future<Item = (), Error = ()> + Send + 'static,
     {
-        let Main {
-            proxy_parts,
-            mut runtime,
-        } = self;
-
-        let (drain_tx, drain_rx) = drain::channel();
-
-        let d = drain_rx.clone();
-        runtime.spawn(futures::lazy(move || {
-            proxy_parts.build_proxy_task(d);
-            trace!("main task spawned");
-            Ok(())
-        }));
-
-        let shutdown_signal = shutdown_signal.and_then(move |()| {
-            debug!("shutdown signaled");
-            drain_tx.drain()
-        });
-
-        runtime.run_until(shutdown_signal).expect("executor");
-
-        debug!("shutdown complete");
+        let Main { proxy_parts } = self;
+        Runtime::new()
+            .expect("runtime")
+            .block_on(
+                futures::lazy(move || {
+                    let (drain_tx, drain_rx) = drain::channel();
+
+                    proxy_parts.build_proxy_task(drain_rx);
+                    trace!("main task spawned");
+
+                    shutdown_signal.and_then(move |()| {
+                        debug!("shutdown signaled");
+                        drain_tx.drain().map(|()| debug!("shutdown complete"))
+                    })
+                })
+                .instrument(info_span!("main")),
```

Comment:
shouldn't do this...

Reply:
?

Reply:
i don't think i actually want `main:` prefixed on all of the log messages

Reply:
ah, yeah. probably get rid of this...